examples/tutorials/mnist_pytorch/README.md

```diff
@@ -1,27 +1,46 @@
 tutorial](https://github.com/pytorch/examples/tree/master/mnist).
 
 ## Files
-* **model_def.py**: The core code for the model. This includes building and compiling the model.
-* **data.py**: The data loading and preparation code for the model.
-* **layers.py**: Defines the convolutional layers that the model uses. 
+* **model.py**: The model definition and custom layers.
+* **data.py**: Code for fetching and pre-processing data for the model.
+* **train.py**: Implementation of the core training loop workflow, the entrypoint for training the model.
 
 ### Configuration Files
 * **const.yaml**: Train the model with constant hyperparameter values.
 * **distributed.yaml**: Same as `const.yaml`, but trains the model with multiple GPUs (distributed training).
+* **dist_random.yaml**: Distributed training with a random grid search algorithm.
 * **adaptive.yaml**: Perform a hyperparameter search using Determined's state-of-the-art adaptive hyperparameter tuning algorithm.
 
 ## Data
-The current implementation uses MNIST data downloaded from AWS S3.
+This examples uses the MNIST dataset from the `torchvision` datasets subpackage. See 
+[torchvision docs](https://pytorch.org/vision/main/generated/torchvision.datasets.MNIST.html#torchvision.datasets.MNIST) 
+for details.
 
 ## To Run
 If you have not yet installed Determined, installation instructions can be found
 under `docs/install-admin.html` or at https://docs.determined.ai/latest/index.html
 
-Run the following command: `det -m <master host:port> experiment create -f 
-const.yaml .`. The other configurations can be run by specifying the appropriate 
+The training loop is invoked through the 
+[`Trainer.fit()`](https://docs.determined.ai/latest/reference/training/api-pytorch-reference.html#determined.pytorch.Trainer.fit) 
+method, which accepts various arguments for configuring training behavior. The example shown in `train.py` 
+illustrates a single implementation that can run in two modes (local training and on-cluster) without
+any code changes.
+
+### Local Training
+The training code in `train.py` can be invoked locally as a regular Python script. Configure the appropriate training 
+lengths, checkpoint/validation periods, or other desired local training functionality in the `Trainer.fit()` call, 
+then run `python3 train.py` from your local environment.
+
+### On-cluster
+To run training on-cluster, configure the desired training arguments (checkpoint or validation periods, 
+checkpoint to start from, etc.) in the `Trainer.fit()` call. An experiment configuration file is also required 
+for on-cluster experiments (several examples are included in the directory).
+
+Then the code can be submitted to Determined for on-cluster training by running this command from the current directory:
+`det experiment create const.yaml .`. The other configurations can be run by specifying the desired 
 configuration file in place of `const.yaml`.
 
```

Comment:
Since you are steering people towards torch.distributed (a decision I agree with), you might want to document what entrypoint changes are needed there.

Also, I don't actually know: does det.launch.torch_distributed work with slots_per_trial=1?

Reply:
added a section for experiment config changes. 

> det.launch.torch_distributed work with slots_per_trial=1?
yeah, it does. don't really want to put it in the `const.yaml` example though cause it seems like a waste to start torch distributed if it's not necessary.

Reply:
At least document that ddp works with a single slot too. "can I have one entrypoint to rule them all" is like an immediate question for the end user, I think.